tests/test_mock_sampler.py

```diff
@@ -37,14 +38,26 @@ def test_properties_and_params(self):
             raise unittest.SkipTest("no QPU available")
 
         mock = MockDWaveSampler()
-
-        self.assertEqual(set(mock.properties), set(sampler.properties))
-        self.assertEqual(set(mock.parameters), set(sampler.parameters))
-
-        #Check extraction of nodelist, edgelist and properties from 
+        # Some properties/parameters like vyfc are harmlessly removed.
+        # Some newer generation processor properties/parameters are harmlessly
+        # removed.
+        # Sufficient to warn, so downgrade from assertion. Note that
+        # this test only runs in the presence of a client connection anyway.
+        comp = {'properties': {'MockDWaveSampler': set(mock.properties), 'default DWaveSampler': set(sampler.properties)},
+                'parameters': {'MockDWaveSampler': set(mock.parameters), 'default DWaveSampler': set(sampler.parameters)}}
+        for k, v in comp.items():
+            for a,b in [('MockDWaveSampler', 'default DWaveSampler'), ('default DWaveSampler', 'MockDWaveSampler')]:
+                diff = v[a].difference(v[b])
+                if len(diff) > 0:
+                    warnings.warn(f'{k} of {a} absent from {b}: {diff}: '
+                                  'Features are added or deprecated across generations', UserWarning)
```

Comment:
Discrepancy between current online params and mock params has been fixed (today) in #600, but changes you propose make sense. Not sure they'll be visible/intrusive enough to catch though.